modules/aws/ec2-files.go

```diff
@@ -15,85 +16,170 @@ type RemoteFileSpecification struct {
 	RemotePathToFileFilter map[string][]string //A map of the files to fetch, where the keys are directories on the remote host and the values are filters for what files to fetch from the directory. The filters support bash-style wildcards.
 	UseSudo                bool
 	SshUser                string
-	KeyPair                *Ec2Keypair
+	SshOptions             *SshOptions
 	LocalDestinationDir    string //base path where to store downloaded artifacts locally. The final path of each resource will include the ip of the host and the name of the immediate parent folder.
 }
 
+// Specify one of KeyPair, SshAgent, or OverrideSshAgent should be specified
+// Any more (or less) than one will result in error on use
+type SshOptions struct {
```

Comment:
Should this be specified in the `aws` package? Seems like a better fit for the `ssh` package...

Reply:
In fact, the `ssh` package already has most of these fields defined in a struct. Can that be reused?

Reply:
possibly, I can see the value in reuse there (other cloud providers would likely want the same thing)

let me review the code between the two modules in more detail, I'll get back to you. it definitely did feel a bit duplicated when writing it